src/Language/PureScript/AST/Declarations.hs

```diff
@@ -820,6 +828,10 @@ data Expr
   --
   | Hole Text
   -- |
+  -- An unknown name that will be turned into a hint/error during typechecking
+  --
+  | UnknownValue Text
```

Comment:
I think this should be a `Name` so that we can easily keep track of what kind of name it is (i.e. whether it is a data constructor, a normal identifier, or an operator) and surface that in the error message easily.

Reply:
I didn't realize how much `printName` did already. You're right, it is super helpful.